### PR TITLE
WIZZIE-3326 Terra-Box documentation

### PR DESCRIPTION
**Pull request by cjmateos**

### CODE REVIEW

File: pages/prelude/terra-box/index.md [hunk 1]
```diff
@@ -5,24 +5,64 @@ datatable: false
...
-Terra-Box is a [HCL](https://github.com/hashicorp/hcl) based project that defines the full WDP infrastructure to be deployed on Kubernetes.
+Terra-Box is a [HCL](https://github.com/hashicorp/hcl) based project that defines the full WDP infrastructure for deploying it on Kubernetes.
 
 This page describes how the project is organized and explains a few of concepts that you should understand to work with WDP deployments.
 
-## Project structure
+## Terra-Box Definition Tree (TDT)
 
-Terra-box is organized in two main parts:
+As we pointed before, WDP is composed by several microservices or components which work together. On Terra-Box, a set of components is defined by a "Terra-Box Definition Tree" (TDT). It is a way of organizing the Terraform files which defines how a component is deployed on Kubernetes. A TDT is organized on the following folders:
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: `As we pointed before`: I think we should link where have been pointed, or omit this sentence.
`components is defined`: change to `components are defined`

File: pages/prelude/terra-box/index.md [hunk 2]
```diff
@@ -5,24 +5,64 @@ datatable: false
...
 
-## Overriding modules
+* `vars.tf` where the module [Input Variables](https://learn.hashicorp.com/terraform/getting-started/variables) (commonly simply called as **vars**) are declared. Module vars let us to parametrize configuration details of the component. E.g.: The password of the admin user.
 
-## ...
+Most of this components can be deployed on kubernetes in a isolated way but some of them might have dependencies.
+
+### main
+
+In order to deploy many WDP components at the same time, two "high level" modules are defined:
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: Change `high level` to `top level`

File: pages/prelude/terra-box/index.md [hunk 3]
```diff
@@ -5,24 +5,64 @@ datatable: false
...
+* **global** - It includes these components which are considered as part of the WDP core (global components). This components are shared for all organizations and users of that WDP deployment.
+* **user** - It includes all those multi-tenant components which have an instance for each organization or user (user components).
+
+In `main` modules (global and user) we can find a `.tf` file for each component defined in `modules` we want to include in. Each `.tf` file imports the module and configures related input variables.
+
+Although there are several Terraform files in a `main` module, all of them are considered as the same main module.
+
+For example, the `global` module includes some components like kafka, zookeeper or druid. All those components are deployed as part of the same main module.
+
+### templates
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: May be we should explain var file priority, indicating that config has priority over flavour and flavour over static

File: pages/prelude/terra-box/index.md [hunk 4]
```diff
@@ -5,24 +5,64 @@ datatable: false
...
+The previous parts of a TDT are used to define WDP components and group them. The `template` folder is used to set input variables **required by the main modules** though.
+
+There are three kind of input variables templates:
+* **config** - It includes those input variables which should be configured by the WDP administrator before to "apply" it. We can find a var file (.tfvars) for each main module (global.tfvars and user.tfvars).
+* **flavour** - The flavour vars usually include configurations about WDP resources such as cpu, memory or replicas for each component. It can be many flavours which pre-configures this vars. Flavour vars files are organized in `global` and `user`.
+* **static** - The static vars include configurations which depend of the WDP version like the components docker image. We can find a var file (.tfvars) for each main module (global.tfvars and user.tfvars).
+
+Both `flavour` and `static` vars are pre-configured and we should only set the `config` vars. However, if we want to customize some `flavour` or `static` var me should add it to the a `config` var file and to modify it there.
+
+## Terra-Box Project structure
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md:75]: May some schemas would be useful to a better understanding

File: pages/prelude/terra-box/index.md [hunk 5]
```diff
@@ -5,24 +5,64 @@ datatable: false
...
+
+Terra-box is organized in two main folders:
+
+* **Base**: It is a TDT for deploying the base components of WDP i.e. the core elements of WDP. Some of them are Kafka, Zookeeper, Druid, and so on.
+
+* **Platforms**: It includes many TDT where components and configurations for adding support to deploy WDP in many cloud providers (AWS, GCP, Azure, ...) or on-premise environments are defined. Here we can find new components and modifications of `Base` components too.
+
+## Plugins
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md:85]: May be we could explain how Base, Platforms and Plugins TDT are combined when an apply is executed.

**Review by pablocasares — APPROVED** [review]

File: pages/prelude/terra-box/index.md [hunk 6]
```diff
@@ -3,26 +3,97 @@ title: Terra-Box
...
-Terra-Box is a [HCL](https://github.com/hashicorp/hcl) based project that defines the full WDP infrastructure to be deployed on Kubernetes.
+Terra-Box is a [HCL](https://github.com/hashicorp/hcl) based project that defines the full WDP infrastructure for deploying it on Kubernetes.
 
 This page describes how the project is organized and explains a few of concepts that you should understand to work with WDP deployments.
 
-## Project structure
+## Terra-Box Definition Tree (TDT) {#tdt}
 
-Terra-box is organized in two main parts:
+WDP is composed by several microservices or components which work together. On Terra-Box, a set of components is defined by a "Terra-Box Definition Tree" (TDT). It is a way of organizing the Terraform files which defines how a component is deployed on Kubernetes. A TDT is organized on the following folders:
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: microservices or components -> microservices and components

File: pages/prelude/terra-box/index.md [hunk 7]
```diff
@@ -3,26 +3,97 @@ title: Terra-Box
...
+* templates
 
-* **Platforms**: It defines components and configurations which add support to deploy WDP in many cloud providers (AWS, GCP, Azure, ...) or on-premise environments.
+{% include image.html file="terra-box/Terra-BoxTDT.png" alt="Terra-Box TDT" max-width=500 %}
 
-## Modules for WDP components
+### modules
 
-## Main modules
+This is the place where the WDP components are defined. Each component definition is created as a Terraform module and it can be imported and used by a higher level module (defined in "main").
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: defined in -> defined at

File: pages/prelude/terra-box/index.md [hunk 8]
```diff
@@ -3,26 +3,97 @@ title: Terra-Box
...
 
-## Templates
+For instance, Wizz-Vis is a WDP component defined in `modules`. It has two Terraform files:
 
-## Overriding modules
+* `main.tf` where the kubernetes resources which deploys Wizz-Vis are defined. This module creates several kubernetes [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment), a [service](https://kubernetes.io/docs/concepts/services-networking/service/) and a [secret](https://kubernetes.io/docs/concepts/configuration/secret).
 
-## ...
+* `vars.tf` where the module [Input Variables](https://learn.hashicorp.com/terraform/getting-started/variables) (commonly simply called as **vars**) are declared. Module vars let us to parametrize configuration details of the component. E.g.: The password of the admin user.
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: the password of the admin user -> the admin user password

File: pages/prelude/terra-box/index.md [hunk 9]
```diff
@@ -3,26 +3,97 @@ title: Terra-Box
...
+
+Most of this components can be deployed on kubernetes in a isolated way but some of them might have dependencies.
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: Terrabox is not designed to deploy modules independently, so I would avoid this phrase

File: pages/prelude/terra-box/index.md [hunk 10]
```diff
@@ -3,26 +3,97 @@ title: Terra-Box
...
+
+In a few words, a plugin is a TDT that adds new components to a WDP deploy or make changes in the components created by Terra-Box or by another plugin.
+
+## Putting all together
+
+In order to deploy a complete WDP we need to process all the involved components defined in the different [TDT](#tdt) (Base, Platform and Plugins). The [main](#mainmodules) modules and its related [templates](#templates) of all TDT are processed at the same level by `terraform`.
+
+{% include image.html file="terra-box/Terra-BoxProcess.png" alt="Terra-Box Process" max-width=500 %}
+
+However, as we have seen, main modules and templates are not centralize. Thus, to apply `terraform` commands, a pre-processing is needed. This processing is made by Prelude.
```
Comment (arodriguezdlc) [inline pages/prelude/terra-box/index.md]: Link to Prelude section

**Review by arodriguezdlc — APPROVED** [review]

**Pull request merged**